synapse_spamcheck_badlist/bad_list_filter.py

```diff
@@ -77,20 +73,36 @@ def __init__(self, config, api):
         self._can_we_check_links = None
         self._can_we_check_md5 = None
 
+        # Timestamp for the latest pull from the links table (or attempt to pull,
+        # if the links table was empty), as returned by `time.time()`.
+        self._last_checked_links = None
+        self._last_checked_md5 = None
+
     async def can_we_check_links(self) -> bool:
```

Comment:
I think we should structure this as:

1. We add a [looping call](https://twistedmatrix.com/documents/current/api/twisted.internet.task.LoopingCall.html) to the constructor to check for updates every N minutes
2. Schedule a call immediately (via `reactor.callWhenRunning`)
3. Then only fetch the links in `check_event_for_spam` if we hadn't yet fetched the links.

This way the relatively expensive fetching and construction of the ahocorasick object never blocks event sending